dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -87,11 +88,10 @@ export const Discussion = ({
 
 	const palette = decidePalette(format);
 
-	const hasCommentsHash =
-		typeof window !== 'undefined' && window.location.hash === '#comments';
+	const hasCommentsHash = !isServer && window.location.hash === '#comments';
 
 	const handlePermalink = (commentId: number) => {
-		if (typeof window === 'undefined') return false;
+		if (isServer) return false;
```

Comment:
If this component shouldn't run on the server then is it better to wrap the `isServer` check around the component that composes it?

Reply:
Also if we do different first-pass renders on the server and client and then try to hydrate then we might get hydration errors:

https://react.dev/reference/react-dom/client/hydrateRoot#caveats

Reply:
This has caused issues in `GetMatchStats` which can be fixed by #9007